File: 0682/Sesión 07 - arquitectura/SoftInvPersistance/src/main/java/pe/edu/pucp/softinv/dao/AlmacenDAO.java
package pe.edu.pucp.softinv.dao;

import java.util.ArrayList;
import pe.edu.pucp.softinv.model.AlmacenesDTO;


public interface AlmacenDAO {

    public Integer insertar(AlmacenesDTO almacen);
    
    public AlmacenesDTO obtenerPorId(Integer almacenId);
    
    public ArrayList<AlmacenesDTO> listarTodos();
    
    public Integer modificar(AlmacenesDTO almacen);
    
    public Integer eliminar(AlmacenesDTO almacen);
}

File: 0683/Sesión 11 - backend en C#/SoftInv/SoftInvPersistance/DAO/AlmacenDAO.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftInv.Model;

namespace SoftInv.DAO
{
    public interface AlmacenDAO 
    {
        int Insertar(AlmacenesDTO almacen);

        AlmacenesDTO ObtenerPorId(int almacenId);

        BindingList<AlmacenesDTO> ListarTodos();

        int Modificar(AlmacenesDTO almacen);

        int Eliminar(AlmacenesDTO almacen);
    }
}
